beacon-chain/sync/service_test.go

```diff
@@ -0,0 +1,232 @@
+package sync
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"hash"
+	"testing"
+
+	"github.com/prysmaticlabs/prysm/beacon-chain/types"
+	"github.com/prysmaticlabs/prysm/shared/testutil"
+	logTest "github.com/sirupsen/logrus/hooks/test"
+	"golang.org/x/crypto/blake2b"
+)
+
+var testLog = log.WithField("prefix", "sync_test")
+
+type MockNetworkService struct{}
+
+func (ns *MockNetworkService) BroadcastBlockHash(h hash.Hash) error {
+	testLog.Infof("broadcasting hash: %x", h.Sum(nil))
+	return nil
+}
+
+func (ns *MockNetworkService) BroadcastBlock(b *types.Block) error {
+	h, err := b.Hash()
+	if err != nil {
+		return err
+	}
+
+	testLog.Infof("broadcasting block: %x", h.Sum(nil))
+	return nil
+}
+
+func (ns *MockNetworkService) RequestBlock(h hash.Hash) error {
+	testLog.Infof("requesting block: %x", h.Sum(nil))
+	return nil
+}
+
+// MockChainService implements a simplified local chain that stores blocks in a slice
+type MockChainService struct {
+	processedHashes []hash.Hash
+}
+
+func (ms *MockChainService) ProcessBlock(b *types.Block) error {
+	h, err := b.Hash()
+	if err != nil {
+		return err
+	}
+
+	testLog.Infof("forwarding block: %x", h.Sum(nil))
+	if ms.processedHashes == nil {
+		ms.processedHashes = []hash.Hash{}
+	}
+
+	ms.processedHashes = append(ms.processedHashes, h)
+	return nil
+}
+
+func (ms *MockChainService) ContainsBlock(h hash.Hash) bool {
+	for _, h1 := range ms.processedHashes {
+		if bytes.Equal(h.Sum(nil), h1.Sum(nil)) {
+			return true
+		}
+	}
+
+	return false
+}
+
+func TestProcessBlockHash(t *testing.T) {
+	hook := logTest.NewGlobal()
+
+	// set the channel's buffer to 0 to make channel interactions synchronous
+	hashBufferSize = 0
```

Comment:
Are either `hashBufferSize` or `blockBufferSize ` being used anywhere ? Same in the other tests, also its not being initialised anywhere so I am not sure how it can be used

Reply:
Ah, yeah, this is confusing. It was a variable at the top of the file. I changed it to a config value.